[Bug]: CPU Backend: vllm / torch.compile issue with torch > 2.8

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>


### 🐛 Describe the bug

If we run vllm on CPU Backend with `torch==2.9` we hit this inductor error:
```
(EngineCore_DP0 pid=73386)   File "/Users/mgoin/code/venvs/vllm/lib/python3.12/site-packages/torch/_inductor/scheduler.py", line 5345, in _codegen_partitions
(EngineCore_DP0 pid=73386)     self._codegen(partition)
(EngineCore_DP0 pid=73386)   File "/Users/mgoin/code/venvs/vllm/lib/python3.12/site-packages/torch/_inductor/scheduler.py", line 5440, in _codegen
(EngineCore_DP0 pid=73386)     raise AssertionError(f"{type(self)=}")
(EngineCore_DP0 pid=73386) torch._inductor.exc.InductorError: AssertionError: type(self)=<class 'torch._inductor.scheduler.Scheduler'>
```

This is not the case with `torch==2.8` which is the current version used in `vllm/requirements/cpu.txt`
However it's problem for Darwin which uses `torch==2.9` and a blocker to updating to `torch==2.9`

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/vllm/config/compilation.py
+++ b/vllm/config/compilation.py
@@ -8,7 +8,6 @@ from dataclasses import asdict, field
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, ClassVar, Literal
 
-import torch
 from pydantic import TypeAdapter, field_validator
 from pydantic.dataclasses import dataclass
 
@@ -666,7 +665,7 @@ class CompilationConfig:
             and "combo_kernels" not in self.inductor_compile_config
             and "benchmark_combo_kernel" not in self.inductor_compile_config
             # (fixme @boyuan) combo kernel does not support cpu yet.
-            and torch.cuda.is_available()
+            and not current_platform.is_cpu()
         ):
             # use horizontal fusion, which is useful for fusing qk-norm and
             # qk-rope when query and key have different shapes.

--- a/vllm/config/compilation.py
+++ b/vllm/config/compilation.py
@@ -8,6 +8,7 @@ from dataclasses import asdict, field
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, ClassVar, Literal
 
+import torch
 from pydantic import TypeAdapter, field_validator
 from pydantic.dataclasses import dataclass
 
@@ -664,6 +665,8 @@ class CompilationConfig:
             is_torch_equal_or_newer("2.9.0.dev")
             and "combo_kernels" not in self.inductor_compile_config
             and "benchmark_combo_kernel" not in self.inductor_compile_config
+            # (fixme @boyuan) combo kernel does not support cpu yet.
+            and torch.cuda.is_available()
         ):
             # use horizontal fusion, which is useful for fusing qk-norm and
             # qk-rope when query and key have different shapes.

